### PR TITLE
fix(Mailer): Allow to enforce strict email format

### PR DESCRIPTION
**Pull request by susnux**
Alternative to https://github.com/nextcloud/server/pull/45425

But this one instead set the global email verification to strict.

### CODE REVIEW

File: lib/private/Mail/Mailer.php [hunk 1]
```diff
@@ -206,8 +207,10 @@ public function validateMailAddress(string $email): bool {
 			// Shortcut: empty addresses are never valid
 			return false;
 		}
+
+		$strictMailCheck = $this->config->getAppValue('core', 'shareapi_enforce_strict_email', 'yes') === 'yes';
```
Comment (come-nc) [inline lib/private/Mail/Mailer.php:211]: I would rename the option name to not mention shareapi, otherwise good :+1:

Comment (susnux) [reply lib/private/Mail/Mailer.php]: Done!

**Review by come-nc — APPROVED** [review]

**Review by Altahrim — APPROVED** [review]

**Comment by joshtrichards** [comment]
I think this is currently breaking CI in master (at least over at suspicious_login, not sure why it isn't here in server currently):

https://github.com/nextcloud/suspicious_login/actions/runs/9779149403/job/26997666619?pr=887
https://github.com/nextcloud/suspicious_login/actions/runs/9894385660/job/27331763334?pr=889

**Pull request merged**